R/gtrends.R

```diff
@@ -163,14 +170,51 @@ gtrends <- function(query, geo, cat, ch, ...) {
 
 #' @rdname gtrends
 #' @export
-gtrends.default <- function(query, geo, cat, ch, ...) {
+gtrends.default <- function(query, 
+                            geo, 
+                            cat, 
+                            ch, 
+                            res = "week",
+                            start_date = as.Date("2004-01-01"),
+                            end_date = as.Date(Sys.time()),
+                            ...){
 
-  if (missing(geo)) geo <- "all"
+  if (missing(geo)) geo <- ""
   if (missing(cat)) cat <- "0"
   if (missing(ch))  ch  <- .getDefaultConnection()
 
   stopifnot(is.character(query),
-            is.vector(query))
+            is.vector(query),
+            all(res %in% c("week", "day")),
+            length(res) == 1)
+  
+  ## Verify the date
+  regex <- "^(19|20)\\d\\d[- /.](0[1-9]|1[012])[- /.](0[1-9]|[12][0-9]|3[01])$"
+
+  if(!grepl(regex, start_date)){
```

Comment:
I am on a religious war with people (cf StackOverflow) who use regexps to validate dates.  That is what a date parser is for!  If you get NA from, fail.  You still pass Feb 30 or for that matter April 31 here.